crates/testing/src/predicates/event.rs

```diff
@@ -58,18 +58,22 @@ where
     TYPES: NodeType,
 {
     let info = format!("{:?}", events);
-    let mut set: HashSet<_> = events.into_iter().collect();
-
-    let function = move |e: &Arc<HotShotEvent<TYPES>>| match set.take(e.as_ref()) {
-        Some(_) => {
-            if set.is_empty() {
-                PredicateResult::Pass
-            } else {
-                PredicateResult::Incomplete
+
+    // Remove duplicate events.
+    let mut events = events;
+    events.dedup();
```

Comment:
I don't think we want to de-duplicate events (this leads to confusing behavior when calling `all(..)`).

Reply:
Sounds good. The HashSet had automatically de-duped so I was trying to find the correct one.